bridge/scripts/lib/madnetFactoryTasks.ts

```diff
@@ -45,53 +78,194 @@ export type Args = {
+    await checkUserDirPath(taskArgs.outputFolder);
+    const path =
+      taskArgs.outputFolder === undefined
+        ? DEFAULT_CONFIG_OUTPUT_DIR
+        : taskArgs.outputFolder;
+    let deploymentList: DeploymentList;
+    let deploymentArgs: DeploymentArgs = {
+      constructor: {},
+      initializer: {},
+    };
+    let list: Array<string>;
+    // no custom path and list input/ writes arg template in default scripts/base-files/deploymentArgs
+    if (
+      taskArgs.outputFolder === undefined &&
+      taskArgs.contractNames === undefined
+    ) {
+      // create the default list
+      // setting list name will specify default configs
+      const contracts = await getAllContracts(hre.artifacts);
+      deploymentList = await getSortedDeployList(contracts, hre.artifacts);
+      list = await transformDeploymentList(deploymentList);
+      deploymentArgs = await generateDeployArgTemplate(list, hre.artifacts);
+    } // user defined path and list
+    else if (
+      taskArgs.outputFolder !== undefined &&
+      taskArgs.contractNames !== undefined
+    ) {
+      // create deploy list and deployment arg with the specified output path
+      const nameList: Array<string> = taskArgs.contractNames;
+      const contracts: Array<string> = [];
+      for (const name of nameList) {
+        const sourceName = (await hre.artifacts.readArtifact(name)).sourceName;
+        const fullName = sourceName + ":" + name;
+        // this will cause the operation to fail if deployType is not specified on the contract
+        await getDeployType(fullName, hre.artifacts);
+        contracts.push(fullName);
+      }
+      deploymentList = await getSortedDeployList(contracts, hre.artifacts);
+      list = await transformDeploymentList(deploymentList);
+      deploymentArgs = await generateDeployArgTemplate(list, hre.artifacts);
+    } // user defined path, default list
+    else if (
+      taskArgs.outputFolder !== undefined &&
+      taskArgs.contractNames === undefined
+    ) {
+      const contracts = await getAllContracts(hre.artifacts);
+      deploymentList = await getSortedDeployList(contracts, hre.artifacts);
+      list = await transformDeploymentList(deploymentList);
+      deploymentArgs = await generateDeployArgTemplate(list, hre.artifacts);
+    } else {
+      throw new Error(
+        "you must specify a path to store your custom deploy config files"
+      );
+    }
+    if (taskArgs.args !== true) {
+      await writeDeploymentList(list, path);
+    }
+    if (taskArgs.list !== true) {
+      await writeDeploymentArgs(deploymentArgs, path);
+      console.log(
+        `YOU MUST REPLACE THE UNDEFINED VALUES IN ${path}/deploymentArgsTemplate`
+      );
+    }
+  });
+
+task("deployContracts", "runs the initial deployment of all madnet contracts")
+  .addFlag(
+    "deployFactory",
+    "flag to indicate deployment, will deploy the factory first if set"
+  )
+  .addOptionalParam("inputFolder", "location for inputs")
```

Comment:
Can you add a new input parameter where the user can specify an address to a factory that was already deployed? We should attach a Factory instance to that address and deploy the contracts using it (maybe even check if that address is a factory before trying to do the deployments)